portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -134,4 +131,18 @@ private int getOffset(HttpServletRequest request) {
       int offset = (currPage - 1) * commentsToShow;
       return offset;
   }
+
+    public String getNickname(String id) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Can't you use the class's `datastore` object rather than re-instantiating it?

Reply:
fixed!